pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/BaseSingleTreeBuilder.java

```diff
@@ -131,7 +132,7 @@ static class Record {
           "Dimension: " + dimension + " does not have dictionary");
     }
 
-    Set<AggregationFunctionColumnPair> functionColumnPairs = builderConfig.getFunctionColumnPairs();
+    Set<AggregationFunctionColumnPair> functionColumnPairs = new TreeSet<>(builderConfig.getFunctionColumnPairs());
```

Comment:
This extra map is redundant as the builder config already create the pairs as `TreeSet`

Reply:
Good catch. Verified all instances are created with TreeSet. Updated